### PR TITLE
Spree2 - Bulk line items controller spec fix

### LINKED ISSUE #2573: [Spree Upgrade] Fix bulk_line_item_controller spec (by defining a shipment in test line_items)
Part of migrating shipping method to shipments, see larger scope in #2009.
Related to the move of inventory_unit.increase to stock_locations and shipments in #2559 
To fix the spec **bulk_line_items_controller_spec**, we need to provide the target_shipment in the test line_items.

The failed test message is:
 ```
1) Admin::BulkLineItemsController#index as a normal user should deny me access to the index action
     Failure/Error: let!(:line_item1) { FactoryBot.create(:line_item, order: order1) }
     NoMethodError:
       undefined method `stock_location' for nil:NilClass
     # /Users/luisramos/dev/ofn/spree/core/app/models/spree/order_inventory.rb:68:in `add_to_shipment'
     # /Users/luisramos/dev/ofn/spree/core/app/models/spree/order_inventory.rb:25:in `verify'
     # /Users/luisramos/dev/ofn/spree/core/app/models/spree/line_item.rb:90:in `update_inventory'
```

This is due to a new hook on spree line_item model: **before_save :update_inventory** that required a shipment to be available either at the line_item level (target_shipment, the one used in the solution, see PR for details) or at inventory level. See method OrderInventory.determine_target_shipment (this method is executed on line_item save, if there is no line_item.target shipment).

I believe this will have to be done in other specs... maybe we create other issues for that.

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Closes #2573 
This means 21 freshly fixed tests in the spree branch build.
See issue for details. 

#### What should we test?
The spec should be green.  

#### Release notes
No release notes.

#### How is this related to the Spree upgrade?
This is part of the spree upgrade, see issue.

### CODE REVIEW

File: app/models/spree/shipment_decorator.rb [hunk 1]
```diff
@@ -2,6 +2,7 @@ module Spree
   Shipment.class_eval do
     def ensure_correct_adjustment_with_included_tax
       ensure_correct_adjustment_without_included_tax
+      return unless adjustment
```
Comment (luisramos0) [inline app/models/spree/shipment_decorator.rb]: this is the "shipment decorator to allow no adjustments in the shipment"

**Comment by luisramos0** [comment]
Yes, it's weird to remove something knowing you will need to put it back at some point in some way. But look, "scoping variants" is a rather important and I am sure we will not forget, anyway, I have created this wiki page for stuff related to that topic:
https://github.com/openfoodfoundation/openfoodnetwork/wiki/Spree-Upgrade:-Scoping-Variants-Products-to-Hubs-and-VariantOverrides

**Comment by luisramos0** [comment]
Yeah, that's a good point Matt. I was planning to leave #2559 open for that, but as you say, I think it's best to close #2559 with this PR and I'll open a new issue just for the scoping part (this way it will be easier for someone else to understand and try this task).

**Comment by daniellemoorhead** [comment]
Semaphore failing... ☹️

**Comment by luisramos0** [comment]
@daniellemoorhead above "into openfoodfoundation:2-0-stable" means this is spree upgrade branch. That means sempahore is on the 75% green mark we talked about: progress. Green in semaphore is only at the 100%.
This issue moves the mark from 75% to 74,9% :-)

**Comment by mkllnk** [comment]
@luisramos0 You mean from 75% to 75.1%? It's a bit confusing that we measure in failed tests, but want to report passed tests to the team. Maybe we should agree on one metric. I find the % passed more positive. :-)

**Comment by luisramos0** [comment]
Yeah, lets forget the failing rate and keep with the success rate. Yes, 75,1% :-)

**Pull request merged**